### PR TITLE
The Colonial Liberation Front Nightmare Inserts

### PR DESCRIPTION
**Pull request by PalJohnson**
## About The Pull Request

This update hopes to introduce colonial liberation into standard games through nightmare inserts survivors. The update was done with the consultation of staff and development as to the best way to re-introduce hostile survivors. This should be the first step in accomplishing this.

## Why It's Good For The Game

First a little back story. This update grew after a short test conducted by staff and in which they introduced hostile survivors. The test initially had survivors choose to either be friendly, neutral, or hostile towards the Marine forces once they landed on the colony. Although the initial test yielded some results as to how to police bad role-playing behavior. I believe more can be done as to the successful implementation of hostile survivors.  In almost all the reports about end proper escalation, the determining Factor seems to revolve around failing to identify which survivor was hostile resulting in antagonistic interaction between the two parties even when the Survivor was declaring themselves as neutral or friendly. 

To resolve this I create these various nightmare inserts for each of the currently playable Maps and end the game to help remedy this issue by giving hostile survivors proper roles that could be easily identified. It was not enough to simply add clf survivors to the normal roster so I decided to add lore-friendly inserts to all of the playable maps to help provide context as to why the CLF is at the colony in the first place.

Much like private military contractor Survivor rules, the colonial Liberation Front Survivor role will only have a 25% chance of spawning on all maps, and when they do spawn they will have a small base of operation to loot for supplies and Stage their Valiant last stand against Zeno's or submarines once they do arrive. The inserts were initially modeled after the crashed ship insert from the LV-624 map As it was the only example of a hostile survivor insert existing.  after discussion with development team members in particular we managed to agree upon design details and proper balance for such hostile forces in an exceedingly rare insert so as to not upset the current flow of the round.

 In order to help facilitate the new survivor roll in its intended purpose several additions needed to be made. The addition of corpses, specifically CLF corpses that could be looted, coding of various objects such as hacked mines with CLF IFF, and the creation of inserts in suitable locations as to not favor marines or Xenos should they take refuge. In addition to the CLF update, I also added in an additional neutral survivor holdout on the map of Shiva's snowball. It was initially slated for another project that never came to pass and I thought why not put it in here.

## Methodology

Initially, the inserts were balanced around the crashed ship insert on the L V6 24 map.  this provided the initial basis for what you see today. The part way during the creation of these inserts I was working on the big red Last Stand update which had a similar map. what is the most common feedback I received from that update was that, unlike the LV counterpart, these Maps were not in an isolated section of the map which had very little to no chance of survival which is why they had an abundance of weapons and ammunition as it was unlikely that they the survivors would be unable to forage additional gear? With that in mind, I made changes to the pre-existing inserts in addition to the big red last add update to help reflect these requests. The map has much of the same gear as the inserts with the exception of turrets add event weapons like the M60 or the UPP specialist sniper rifle.

The locations of the maps were selected for two main reasons. The first reason was to help provide an engaging story or a reasonable area in which a firefight of sorts would take place should a clone of Liberation Front cell attack The Colony or the colonists themselves rise up. Secondly, the maps or areas selected had to be far away from The Landing zones and from the hive as to not have a considerable impact on the flow of the rounds.  This understandably made the selection very difficult with various different buildings across all the colonies had received some sort of insert until one was picked.  had I placed a clf hold out nearby Landing Zone then the direct conflict between Marines and hostile forces would happen immediately disrupting the start of the round. Not to mention the fact that loot would be immediately available to the marines. Had I put the insert closer to Hive locations then Xenomorphs would have to attack the fortified clf soldiers at an early stage or have to relocate as a result. These locations should provide for a happy medium between all forces at play.

## Maps

**_Florina Science Annex Crashed Ship Insert_**

- I know what you are thinking. Yes this is the crashed ship insert from LV posted here with some minor alterations. Initial credits for ship go to benedict. I did this because the map was too good not to use and there were really no go areas to stage a clf invasion that wouldn't be a meta pick for survivors, marines, or Xenos. So I decided to crash the CLF ship once more onto the northern side of the prison station where the crew of the prison attempted to thwart the invasion. Both side have taken many casualties and here you play as recently awaken CLF survivors who managed to sleep threw the ordeal. Will you fight to take over the prison or will you wait to call for backup? The choice is yours.

**_Soro Map Insert_**

- In this insert, the political prisoners of the colony have taken arms against the UPP and WY overlords and have raided the armory for weapons. After a successful fight they managed to take over the security department and have established themselves as a new cell of the colonial liberation front with a vendetta not just for the corporation but also the  UPP.

**_LV-624 Hydroponics CLF Insert_**

- This insert reflects the valiant efforts of the colonist and pmc forces having a last stand in the hydroponics facility of the colony. This location was selected due to its distance between LZ1 and LZ2. In this insert, the CLF have just recently slain the last of the loyalist colonist and their paid guns for hire. Although the map does feature a CLF insert the most common complaint received about it was that the hostile survivors had no way of escaping. Even with their sentries, the clf survivors there had no chance of surviving. This insert should at least give the survivors a fighting chance at the cost of all the gear they would have normally received had they stayed in the ship

**_Kutjevo CLF Survivor Insert_**

- The kutjevo refinery luck seemed to have run out. The colonist banded together to form their very own cell of the colonial liberation front after their calls for better working conditions and pay failed to yield results. In secret the colonist have been amassing a great deal of equipment in order to stage a rebellion when they were caught in the act by local security forces resulting in a shootout at the local bar.

**_Shiva Snowball CLF Insert_**

- In this insert, the CLF forces having received a distress signal from the shivas snowball facility dispatched a small cadre of foot soldiers to take over the guarded facility. An intense firefight broke out at the administration section of the colony in which security forces, colonist, and their cowardly administrators barricaded themselves in sending a distress signal of their own which would be eventually received by the marines. In this insert, CLF survivors play as the remaining survivors of the battle and have established themselves as the new authority of the colony.

**_New Survivor Neutral Last Stand Insert_**

This map was supposed to be a part of an extended last-stand update for all maps however given the number of projects I had I was unable to complete it to the fullest. It didn't help that my repo corrupted on multiple occasions. I managed to find a backup of the map and decide to insert it here having updated it with the Big Red last stand insert in mind. Here survivors have barricaded themselves in the colony bar in the hopes they are to be rescued after having sent a distress signal. Only time will tell if help arrives.

## Changelog

:cl: theflyingflail
add: Added new clf corpses to help decorate maps with
add: Added new clf survivor spawner with updated text
add: Added new clf mines at the request of Geeves
add: Added new clf inserts to all currently playable maps
add: Added new survivor last stand map to Shiva snowball

/:cl:

### CODE REVIEW

**Comment by JediKing112** [comment]
how would you deal with, for instance, the CLF survivor on LV shooting from west nexus and running away before they can be caught? there are a lot of hiding spots especially for somebody with a powerful ranged weapon that can be fired from off-map into high density areas

**Comment by PalJohnson** [comment]
> how would you deal with, for instance, the CLF survivor on LV shooting from west nexus and running away before they can be caught? there are a lot of hiding spots, especially for somebody with a powerful ranged weapon that can be fired from off-map into high-density areas

That would be dealt with by staff. It's the same as Xenos attacking LZ on the first drop. Xenos needs to wait for the marines to drop at least two times before attacking LZ. To my knowledge, there are no rifles other than the hunting rifle which has a scope. That weapon in particular isn't even impressive. I could remove scoped weapons from inserts this proves to be the problem.

**Review by Agameofscones — CHANGES_REQUESTED** [review]
Do not use legacy doors on the Sci-Annex insert, use Almayer variant doors with null access.

Incorrect window frame used on Shiva's insert.

**Comment by c4xmaniac** [comment]
Such an awesome MR, thank you Pal.

**Comment by PalJohnson** [comment]
> Do not use legacy doors on the Sci-Annex insert, use Almayer variant doors with null access.
> 
> Incorrect window frame used on Shiva's insert.

Made changes requested

File: maps/map_files/DesertDam/Desert_Dam.dmm [hunk 1]
```diff
@@ -65095,9 +65095,6 @@
 	dir = 8
```
Comment (Nanu308) [inline maps/map_files/DesertDam/Desert_Dam.dmm:65095]: Trijent Dam:
- Noop turfs and areas when there are no changes between your insert in certain locations and the base map itself.
- bilde
- bilde
- What happened to all the weapons and armory? Did the CLF troopers just not use them? It looks to clean in this armor, also, the walls are too strong around the armory. Drop the guns around the area on the corpses or nearby, and limit the manually spawned guns if these corpses spawn with guns. Also remember CLF are in general underfunded and lack good equipment.
- bilde
- Change the Health on the supposed damaged barricades or they'll just look damaged till they take dmg and change their sprite to be fine again.
- bilde
- Chill on the CLF posters, a few here and there is fine, don't need as many as you currently have.

File: maps/map_files/FOP_v3_Sciannex/Fiorina_SciAnnex.dmm [hunk 2]
```diff
@@ -30036,6 +30036,9 @@
 /area/fiorina/station/power_ring)
 "sgb" = (
 /obj/item/ammo_box/magazine/misc/flares/empty,
+/obj/effect/landmark/nightmare{
```
Comment (Nanu308) [inline maps/map_files/FOP_v3_Sciannex/Fiorina_SciAnnex.dmm:30039]: - You're copy pasting a massive piece/the entire LV CLF ship insert onto Fiorina. It's current location is not very good and remember ship hull/walls can not be broken/torn down meaning there is no flanking routes or different ways to enter it than the south side.
- Same as before, use Noop turf/area when there are no changes between a small area on the nightmare insert and base map.
- Why are there so many invis walls up North on the blocked off part? If you're worried about burrowers etc. check the area code for the space there, see if they can burrow as i believe they already cant.
- Tldr. Re-do it in its entirety or drop this change.

File: maps/map_files/Ice_Colony_v3/Shivas_Snowball.dmm [hunk 3]
```diff
@@ -15887,6 +15887,12 @@
 	},
 /turf/open/auto_turf/snow/layer1,
 /area/shiva/exterior/cp_colony_grounds)
+"kCQ" = (
```
Comment (Nanu308) [inline maps/map_files/Ice_Colony_v3/Shivas_Snowball.dmm:15890]: - Same as before, use Noop turf/area when there are no changes between a small area on the nightmare insert and base map.
- Why are the walls changing color? Did the CLF decide to build new grey walls on the outside near the offices?
- There is a HPR still inside of this office, did the CLF not search around or use it etc.?
- bilde
- Too many posters again, touch it down a little.
- Damaged barricades needs their health VV'd as well.
- Calm down on the landmarks for spawners, this isn't the Cryocells on-board the USS Almayer, delete a fair few, consider locations where survivors would "wake" up or be standing guard etc.
- Calm down on the ammo for the CLF here as well, remember, they have been fighting and are already underfunded etc.
- bilde

File: maps/map_files/Ice_Colony_v3/sprinkles/40.shivabarls.dmm [hunk 4]
```diff
@@ -0,0 +1,1959 @@
+//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE
+"ak" = (
```
Comment (Nanu308) [inline maps/map_files/Ice_Colony_v3/sprinkles/40.shivabarls.dmm:2]: - Generate Icons by Icon State and place those etc.
- bilde
- A lot of object clutter.
- CLF equipment somehow better equipped than the UPP on the bar with multiple medkits, weapons and ammo. 
- The colored parkas are cool.
- Damaged cades and health, you know what to do.
- Noop turfs/area as previously said in above comments.
- The dog tag bottom left has no information, give it some information for that sweet lore.

File: maps/map_files/Kutjevo/Kutjevo.dmm [hunk 5]
```diff
@@ -8838,6 +8838,12 @@
 /obj/structure/largecrate/random/case/double,
 /turf/open/floor/kutjevo/colors/orange,
 /area/kutjevo/exterior/Northwest_Colony)
+"mfs" = (
```
Comment (Nanu308) [inline maps/map_files/Kutjevo/Kutjevo.dmm:8841]: - Same as before, use Noop turf/area when there are no changes between a small area on the nightmare insert and base map.
- Damaged Barricades and Health.
- Ammunition and Guns, too much plus too collected to reflect a last stand area. Spread them out among corpses etc.
- Way too many spawn landmarks, as before spread them out a bit more and less.

**Review by Nanu308 — CHANGES_REQUESTED** [review]
This is what i have time for tonight. Will review more another time.
